content/public/browser/guest_frame.h

```diff
@@ -22,15 +27,26 @@ class WebContents;
 // to align with SecureEmbedDelegate's naming pattern.
 class CONTENT_EXPORT GuestFrame {
  public:
+  // Observer interface for GuestFrameImpl events that are forwarded to the
+  // embedder.
+  class GuestFrameObserver : public base::CheckedObserver {
```

Comment:
I think GuestFrame won't have a second Observer, so a Client or Delegate class will be more appropriate.